components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -4237,6 +4237,29 @@ public static String getAccessTokenIdentifier(OAuth2TokenValidationResponseDTO t
         return null;
     }
 
+    /**
+     * Return access token DO from OAuth2TokenValidationResponseDTO. This method validated the token against DB or
+     * in-memory.
+     *
+     * @param tokenResponse OAuth2TokenValidationResponseDTO object.
+     * @return extracted access token DO.
+     * @throws UserInfoEndpointException if  error occurred while obtaining access token.
+     */
+    public static AccessTokenDO getAccessTokenDO(OAuth2TokenValidationResponseDTO tokenResponse)
```

Comment:
Lets return a optional as there is a possibility of null being returned.

ref: https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2176/files#r1366476831

Reply:
fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2176/commits/ae287fa0b87cd0e48f8fd62b035ca1a6e7fca706